### ISSUE TITLE
failed assertion `[MPSNDArrayDescriptor sliceDimension:withSubrange:] error: subRange.start (6) is not less than length of dimension[0] (6)'

### ISSUE THREAD

**Issue by shink** [pytorch/pytorch#125832]
### 🐛 Describe the bug

I got an error when running this test. This issue looks similar to #96153, but it appears again.

```bash
python test/test_mps.py
```

output:

```
........./opt/anaconda3/envs/pth/lib/python3.11/site-packages/torch/nn/functional.py:4539: UserWarning: MPS: The constant padding of more than 3 dimensions is not currently supported natively. It uses View Ops default implementation to run. This may have performance implications. (Triggered internally at /Users/shenke/workspace/pytorch/aten/src/ATen/native/mps/operations/Pad.mm:472.)
  return torch._C._nn.pad(input, pad, mode, value)
........F......./Users/shenke/workspace/pytorch/test/test_mps.py:9859: UserWarning: The use of `x.T` on tensors of dimension other than 2 to reverse their shape is deprecated and it will throw an error in a future release. Consider `x.mT` to transpose batches of matrices or `x.permute(*torch.arange(x.ndim - 1, -1, -1))` to reverse the dimensions of a tensor. (Triggered internally at /Users/shenke/workspace/pytorch/aten/src/ATen/native/TensorShape.cpp:3689.)
  t1 = a.T
./Users/shenke/workspace/pytorch/test/test_mps.py:9229: UserWarning: TypedStorage is deprecated. It will be removed in the future and UntypedStorage will be the only storage class. This should only matter to you if you are using storages directly.  To access UntypedStorage directly, use tensor.untyped_storage() instead of tensor.storage()
  if base.storage().data_ptr() != other.storage().data_ptr():
...................../AppleInternal/Library/BuildRoots/ce725a5f-c761-11ee-a4ec-b6ef2fd8d87b/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:124: failed assertion `[MPSNDArrayDescriptor sliceDimension:withSubrange:] error: subRange.start (6) is not less than length of dimension[0] (6)'
[1]    55309 abort      python test/test_mps.py
```

### Versions

```bash
$ python torch/utils/collect_env.py
Collecting environment information...
PyTorch version: 2.4.0a0+gitfc2256f
Is debug build: True
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 14.4.1 (arm64)
GCC version: Could not collect
Clang version: 15.0.0 (clang-1500.3.9.4)
CMake version: version 3.26.4
Libc version: N/A

Python version: 3.11.9 (main, Apr 19 2024, 11:43:47) [Clang 14.0.6 ] (64-bit runtime)
Python platform: macOS-14.4.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M3 Pro

Versions of relevant libraries:
[pip3] flake8==7.0.0
[pip3] flake8-bugbear==23.3.23
[pip3] flake8-comprehensions==3.12.0
[pip3] flake8-executable==2.1.3
[pip3] flake8-logging-format==0.9.0
[pip3] flake8-pyi==23.3.1
[pip3] flake8-simplify==0.19.3
[pip3] mypy==1.9.0
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.26.0
[pip3] optree==0.11.0
[pip3] torch==2.4.0a0+gitfc2256f
[conda] numpy                     1.26.0                   pypi_0    pypi
[conda] optree                    0.11.0                   pypi_0    pypi
[conda] torch                     2.4.0a0+gitfc2256f          pypi_0    pypi
[conda] torchfix                  0.4.0                    pypi_0    pypi
```

cc @seemethere @malfet @pytorch/pytorch-dev-infra @kulinseth @albanD @DenisVieriu97 @jhavukainen

**Comment by malfet** [comment]
Do you know which test are you encountering it with?

**Comment by shink** [comment]
@malfet Thanks for your guidance. The result of rerunning the test with `-v`:

```
test_contiguous_nonview (__main__.TestViewOpsMPS.test_contiguous_nonview) ... ok
test_contiguous_self (__main__.TestViewOpsMPS.test_contiguous_self) ... ok
test_detached_view_copy (__main__.TestViewOpsMPS.test_detached_view_copy) ... ok
test_diagonal_view (__main__.TestViewOpsMPS.test_diagonal_view) ... ok
test_empty_reshape (__main__.TestViewOpsMPS.test_empty_reshape) ... ok
test_expand (__main__.TestViewOpsMPS.test_expand) ... ok
test_expand_as_view (__main__.TestViewOpsMPS.test_expand_as_view) ... ok
test_expand_view (__main__.TestViewOpsMPS.test_expand_view) ... ok
test_flatten_nonview (__main__.TestViewOpsMPS.test_flatten_nonview) ... ok
test_flatten_view (__main__.TestViewOpsMPS.test_flatten_view) ... ok
test_inplace_view_add (__main__.TestViewOpsMPS.test_inplace_view_add) ... /AppleInternal/Library/BuildRoots/ce725a5f-c761-11ee-a4ec-b6ef2fd8d87b/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:124: failed assertion `[MPSNDArrayDescriptor sliceDimension:withSubrange:] error: subRange.start (6) is not less than length of dimension[0] (6)'
[1]    72470 abort      python test/test_mps.py -v
```

Run this single test:

```bash
$ python test/test_mps.py -k TestViewOpsMPS.test_inplace_view_add                   
/AppleInternal/Library/BuildRoots/ce725a5f-c761-11ee-a4ec-b6ef2fd8d87b/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:124: failed assertion `[MPSNDArrayDescriptor sliceDimension:withSubrange:] error: subRange.start (6) is not less than length of dimension[0] (6)'
[1]    76320 abort      python test/test_mps.py -k TestViewOpsMPS.test_inplace_view_add
```

### versions:

```bash
$ python torch/utils/collect_env.py
Collecting environment information...
PyTorch version: 2.4.0a0+gitfc2256f
Is debug build: True
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 14.4.1 (arm64)
GCC version: Could not collect
Clang version: 15.0.0 (clang-1500.3.9.4)
CMake version: version 3.26.4
Libc version: N/A

Python version: 3.11.9 (main, Apr 19 2024, 11:43:47) [Clang 14.0.6 ] (64-bit runtime)
Python platform: macOS-14.4.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M3 Pro

Versions of relevant libraries:
[pip3] flake8==7.0.0
[pip3] flake8-bugbear==23.3.23
[pip3] flake8-comprehensions==3.12.0
[pip3] flake8-executable==2.1.3
[pip3] flake8-logging-format==0.9.0
[pip3] flake8-pyi==23.3.1
[pip3] flake8-simplify==0.19.3
[pip3] mypy==1.9.0
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.26.0
[pip3] optree==0.11.0
[pip3] torch==2.4.0a0+gitfc2256f
[conda] numpy                     1.26.0                   pypi_0    pypi
[conda] optree                    0.11.0                   pypi_0    pypi
[conda] torch                     2.4.0a0+gitfc2256f          pypi_0    pypi
[conda] torchfix                  0.4.0                    pypi_0    pypi
```

**Comment by DenisVieriu97** [comment]
@shink thanks for the update.

> PyTorch version: 2.4.0a0+gitfc2256f

I see that you're using a locally built torch. Could you please try a nightly version and try again?
```
pip3 install --pre --upgrade torch --index-url https://download.pytorch.org/whl/nightly/cpu
```
then run
```
python test/test_mps.py -k test_inplace_view_add -v
```

**Comment by shink** [comment]
@DenisVieriu97 Thanks a lot! This test works fine on the nightly version. So maybe this issue only occurs on M3Pro? :p

run:

```bash
conda create -n testpth python=3.11
conda activate testpth

pip3 install --pre --upgrade torch --index-url https://download.pytorch.org/whl/nightly/cpu
pip3 install -r requirements.txt

cd workspace/pytorch
python test/test_mps.py -k TestViewOpsMPS.test_inplace_view_add -v
```

output:

```
est_inplace_view_add (__main__.TestViewOpsMPS.test_inplace_view_add) ... ok

----------------------------------------------------------------------
Ran 1 test in 0.069s

OK
```

**Comment by DenisVieriu97** [comment]
Thank you @shink for confirming. The issue was recently fixed (commit [ba275486791622a7493f032d02ad420fa9497d30](https://github.com/pytorch/pytorch/commit/ba275486791622a7493f032d02ad420fa9497d30)) - your previous pytorch build must have been older. 
Please let me know if you see any other issues
